src/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSource.cs

```diff
@@ -7,26 +7,26 @@
 
 namespace System.Diagnostics
 {
-    public partial class DiagnosticListener : System.Diagnostics.DiagnosticSource, System.IDisposable, System.IObservable<System.Collections.Generic.KeyValuePair<string, object>>
+    public partial class DiagnosticListener : System.Diagnostics.DiagnosticSource, System.IDisposable, System.IObservable<System.Collections.Generic.KeyValuePair<string?, object?>>
     {
-        public DiagnosticListener(string name) { }
+        public DiagnosticListener(string? name) { }
         public static System.IObservable<System.Diagnostics.DiagnosticListener> AllListeners { get { throw null; } }
-        public string Name { get { throw null; } }
+        public string? Name { get { throw null; } }
         public virtual void Dispose() { }
         public bool IsEnabled() { throw null; }
         public override bool IsEnabled(string name) { throw null; }
-        public override bool IsEnabled(string name, object arg1, object arg2 = null) { throw null; }
-        public virtual System.IDisposable Subscribe(System.IObserver<System.Collections.Generic.KeyValuePair<string, object>> observer) { throw null; }
-        public virtual System.IDisposable Subscribe(System.IObserver<System.Collections.Generic.KeyValuePair<string, object>> observer, System.Func<string, object, object, bool> isEnabled) { throw null; }
-        public virtual System.IDisposable Subscribe(System.IObserver<System.Collections.Generic.KeyValuePair<string, object>> observer, System.Predicate<string> isEnabled) { throw null; }
-        public override string ToString() { throw null; }
-        public override void Write(string name, object value) { }
+        public override bool IsEnabled(string name, object? arg1, object? arg2 = null) { throw null; }
+        public virtual System.IDisposable Subscribe(System.IObserver<System.Collections.Generic.KeyValuePair<string?, object?>> observer) { throw null; }
```

Comment:
```suggestion
        public virtual System.IDisposable Subscribe(System.IObserver<System.Collections.Generic.KeyValuePair<string, object?>> observer) { throw null; }
```

What happens if we declare a string as being non-nullable but someone's pre-existing code did pass null? Do they have ways to suppress those errors/warnings? This is another case where I would never recommend to pass a null string (it is the name of the event), but technically an instrumented library + diagnostic cool could mutually agree that the null event has a specific meaning and use it.

Reply:
> What happens if we declare a string as being non-nullable but someone's pre-existing code did pass null?

Until they enable nullable warnings in their project, nothing happens, it's exactly the same as it is today.  If they then enable nullable warnings and the value they're passing in may be null, the compiler will warn.

> Do they have ways to suppress those errors/warnings?

Yes.  There are multiple ways the warnings (or warnings-as-errors) can be suppressed.  All of the normal mechanisms work (e.g. `#pragma warning disable`, `NoWarn`, etc.), but you can also use `!` to suppress the failures.

Reply:
Is it fair to say we use the nullable markers as recommendations for what we expect good usage will be, even if the code wouldn't necessarily have failed or rejected a null value?


I started commenting various places below and then stopped after it was becoming very duplicative. It distills down to listener names, event names, tag names, and baggage names are recommended not to be null. In places where we have key value pairs I think it is legitimate to have null for the value, but not recommended for the name component.

Reply:
> Is it fair to say we use the nullable markers as recommendations for what we expect good usage will be, even if the code wouldn't necessarily have failed or rejected a null value?

See https://github.com/dotnet/corefx/blob/master/Documentation/api-guidelines/nullability.md.  We want the annotations to correctly annotate the intention of the API, even if the API isn't explicitly or always validating adherence to the contract.

Reply:
Seems we are getting agreed to make names non null, values nullable

> I started commenting various places below and then stopped after it was becoming very duplicative. It distills down to listener names, event names, tag names, and baggage names are recommended not to be null. In places where we have key value pairs I think it is legitimate to have null for the value, but not recommended for the name component.

Yes they all will be updated accordingly same, if we got agreed to which annotation to use